Ensure invalid username/password returns 401 error, not 403

CaoShuFeng opened the pull request:
If a user attempts to use basic auth, and the username/password combination
is rejected, the authenticator should return an error. This distinguishes
requests that did not provide username/passwrod (and are unauthenticated
without error) from ones that attempted to, and failed.

Related to:
https://github.com/kubernetes/kubernetes/pull/39408

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/41775)
<!-- Reviewable:end -->

CaoShuFeng commented:
@k8s-bot ok to test

CaoShuFeng committed:
Ensure invalid username/password returns 401 error, not 403

If a user attempts to use basic auth, and the username/password combination
is rejected, the authenticator should return an error. This distinguishes
requests that did not provide username/passwrod (and are unauthenticated
without error) from ones that attempted to, and failed.

liggitt commented:
/lgtm

liggitt commented:
/approve

liggitt commented:
looks like lint is unhappy with that var name:
```
W0221 12:24:35.513] staging/src/k8s.io/apiserver/plugin/pkg/authenticator/request/basicauth/basicauth.go:37:5: error var invalidAuth should have name of the form errFoo
W0221 12:24:35.513] 
W0221 12:24:35.514] Please fix the above errors. You can test via "golint" and commit the result.
W0221 12:24:35.514] 
I0221 12:24:35.614] FAILED   hack/make-rules/../../hack/verify-golint.sh	46s
```

CaoShuFeng commented:
@liggitt updated the var name.

CaoShuFeng commented:
@k8s-merge-robot is crazy commit my pr.

liggitt commented:
@apelisse the bot has deleted and added the "not approved" comment about 200 times in the last few hours. Did a bad update just roll out?

apelisse commented:
@liggitt Look like it stopped?

liggitt commented:
@k8s-bot ok to test

liggitt commented:
/lgtm

CaoShuFeng commented:
@k8s-bot non-cri e2e test this

CaoShuFeng commented:
Hi, @deads2k can you add add an approve label for this patch?
